src/ufs.rs

```diff
@@ -148,7 +146,8 @@ impl Ufs {
 }
 
 fn run<T>(f: impl FnOnce() -> IoResult<T>) -> Result<T, c_int> {
-	f().map_err(|e| e.raw_os_error().unwrap_or(libc::EIO))
+	f().inspect_err(|e| eprintln!("Error: {e:?}"))
```

Comment:
We shouldn't be using printf.  Instead, use the log crate's functions, like this:
```suggestion
	f().inspect_err(|e| log::error!!(e))
```